docs/inventory_object.md

```diff
@@ -0,0 +1,194 @@
+  ```yaml
+  <Group1>_<Kind1>_<Namespace1>_<Name1>---<Group2>_<Kind2>_<Namespace2>_<Name2>: <hash>
+  ```
+  For example, a ConfigMap object referred by a Deployment object
+  can be represented as
+  ```yaml
+  _Secret_default_pass-ktm4b8f895---apps_Deployment_default_mysql:  m99cb8h85h
+  ```
+
+Basically, this ConfigMap object acts a record of objects that are applied as a group.
+This ConfigMap object can be consumed by a client such as
+[cli-experimental](https://github.com/kubernetes-sigs/cli-experimental).
+The client can recognize this ConfigMap object and take proper actions
+when running apply, prune and delete.
+
+### Example
+Take following `kustomization.yaml` as an example
+```yaml
+resources:
+- deployment.yaml
+- service.yaml
+
+
+secretGenerator:
+- name: pass
+  literals:
+  - password=secret
+
+inventory:
+  type: ConfigMap
+  configMap:
+    name: root-cm
+    namespace: default
+
+namespace: default
+```
+
+where the `deployment.yaml` is
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: mysql
+  labels:
+    app: mysql
+spec:
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      app: mysql
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: mysql
+    spec:
+      containers:
+      - image: mysql:5.6
+        name: mysql
+        env:
+        - name: MYSQL_ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: pass
+              key: password
+        ports:
+        - containerPort: 3306
+          name: mysql
+        volumeMounts:
+        - name: mysql-persistent-storage
+          mountPath: /var/lib/mysql
+      volumes:
+      - name: mysql-persistent-storage
+        emptyDir: {}
+```
+
+and the `service.yaml` is
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: mysql
+  labels:
+    app: mysql
+spec:
+  ports:
+    - port: 3306
+  selector:
+    app: mysql
+```
+
+Running `kustomize build` gives 4 objects.
+Besides the Deployment `mysql`, the Service `mysql`,
+and the Secret `pass`, the output also contains a
+ConfigMap object as
+```yaml
+apiVersion: v1
+data:
+  _Secret_default_pass-dfg7h97cf6: f5bg5d8k42
+  _Secret_default_pass-dfg7h97cf6---apps_Deployment_default_mysql: f5bg5d8k42
+  _Service_default_mysql: f5bg5d8k42
+  apps_Deployment_default_mysql: f5bg5d8k42
+kind: ConfigMap
+metadata:
+  annotations:
+    current: f5bg5d8k42
+  name: root-cm
+  namespace: default
+```
+
+It is clear that this ConfigMap contains a record of all
+objects built from this Kustomization target.
+
+
+### Hash
+Note that in the ConfigMap generated from `inventory` field, there is a hash
+`f5bg5d8k42`. This hash is added as the value for each key in the data field.
+It is also added to the ConfigMap's annotation as
+```yaml
+metadata:
+  annotations:
+    current: f5bg5d8k42
+```
+
+This hash is computed by hashing all the keys in data field, which is the following list
```

Comment:
hashing all the _keys_ in the inventory.  In the above example, the keys are:

Reply:
nit: in the ConfigMap ... there is a hash `b965tb9c7d`. Which value are you referring to?